sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/ImplementIDisposableCorrectly.AbstractClass.cs

```diff
@@ -0,0 +1,73 @@
+using System;
+
+namespace Tests.Diagnostics
+{
+    public abstract class AbstractNotDisposable
+    {
+        public void Dispose() { } // is ignored
+    }
+
+    public abstract class AbstractWithAbstractDispose : IDisposable // Noncompliant
+    {
+        public abstract void Dispose();
+//             ^^^^^^^^ Secondary {{'AbstractWithAbstractDispose.Dispose()' should not be 'virtual' or 'abstract'.}}
+        protected abstract void Dispose(bool disposing);
+    }
+
+    public abstract class AbstractWithoutProtectedDispose : IDisposable
+//                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Noncompliant {{Fix this implementation of 'IDisposable' to conform to the dispose pattern.}}
+//                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Secondary@-1 {{Provide 'protected' overridable implementation of 'Dispose(bool)' on 'AbstractWithoutProtectedDispose' or mark the type as 'sealed'.}}
+    {
+    public void Dispose() // Secondary {{'AbstractWithoutProtectedDispose.Dispose()' should only invoke 'Dispose(true)' and 'GC.SuppressFinalize(this)'.}}
+        {
+            GC.SuppressFinalize(this);
+        }
+    }
+
+    public abstract class AbstractWithVirtual : IDisposable
+    {
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+        }
+    }
+
+    public class DerivedFromAbstractWithVirtual : AbstractWithVirtual
+    {
+        protected override void Dispose(bool disposing)
+        {
+            base.Dispose(disposing);
+        }
+    }
+
+    public class DerivedFromAbstractWithVirtualNoBaseCall : AbstractWithVirtual // Noncompliant
+    {
+        protected override void Dispose(bool disposing) { }
+//                              ^^^^^^^ Secondary {{Modify 'Dispose(disposing)' so that it calls 'base.Dispose(disposing)'.}}
+    }
+
```

Comment:
That's a bit stupid to do that but please add a test where user will override and call base from an expression body:
`protected override void Dispose(bool disposing) => base.Dispose(disposing);`